ideascube/serveradmin/catalog.py

```diff
@@ -427,26 +440,41 @@ def _progress(*args):
 
     def list_installed(self, ids):
         pkgs = self._get_packages(
-            ids, self._catalog['installed'], fail_if_no_match=False)
+            ids, self._catalog['installed'],
+            fail_if_no_match=False, fail_if_invalid_package=False)
         return sorted(pkgs, key=attrgetter('id'))
 
     def list_available(self, ids):
         pkgs = self._get_packages(
-            ids, self._catalog['available'], fail_if_no_match=False)
+            ids, self._catalog['available'],
+            fail_if_no_match=False, fail_if_invalid_package=False)
         return sorted(pkgs, key=attrgetter('id'))
 
     def list_upgradable(self, ids):
         pkgs = []
 
         for ipkg in self._get_packages(
-                ids, self._catalog['installed'], fail_if_no_match=False):
+                ids, self._catalog['installed'],
+                fail_if_no_match=False, fail_if_invalid_package=False):
             upkg = self._get_package(ipkg.id, self._catalog['available'])
 
             if ipkg != upkg:
                 pkgs.append(upkg)
 
         return sorted(pkgs, key=attrgetter('id'))
 
+    def list_nothandled(self, ids):
```

Comment:
In general I have reservations about silently ignoring anything.

Finding unhandled packages in a catalog you're consuming is a clear sign that something isn't right and ought to be fixed, among others:

* the catalog provider made a typo
* the client is out of date

Maybe listing the unhandled packages isn't the right thing to do, I don't know. Should we instead just display a warning, something like `Warning: Ignoring unhandled package 'foo' ('unhandled-type')` just as the problematic package is encountered? (that is, before reporting the list)

But I would argue that (at least for the moment, while things are still being fleshed out), we really want to know that some packages in the catalog can't be installed, as it's a sign we made a mistake and we'll want to fix it.